apps/bridge/src/utils/hooks/useCCTPBridgeStatus.ts

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useEffect, useState } from 'react';
+import { usePublicClient, useWaitForTransaction } from 'wagmi';
+import getConfig from 'next/config';
+import { CCTPBridgePhase } from 'apps/bridge/src/utils/transactions/phase';
+import { useQuery } from 'react-query';
+import { keccak256, getEventSelector, decodeAbiParameters, parseAbiParameter } from 'viem';
+import MessageTransmitter from 'apps/bridge/src/contract-abis/MessageTransmitter';
+
+const { publicRuntimeConfig } = getConfig();
+const l1ChainID = parseInt(publicRuntimeConfig.l1ChainID);
+const l2ChainID = parseInt(publicRuntimeConfig.l2ChainID);
+
+type BridgeAttestation = {
+  attestation: `0x${string}`;
+  status: string;
+};
+
+async function fetchBridgeAttestation(messageHash: string): Promise<BridgeAttestation | undefined> {
+  const response = await fetch(
+    `${publicRuntimeConfig.cctpAttestationsAPIURL}/attestations/${messageHash}`,
+  );
+
+  return (await response.json()) as BridgeAttestation;
+}
+
+type UseCCTPBridgeStatusProps = {
+  initiateTxHash: `0x${string}`;
+  bridgeDirection: 'deposit' | 'withdraw';
+};
+
+type CCTPBridgeStatus = {
+  status: CCTPBridgePhase;
+  message?: `0x${string}`;
+  attestation?: `0x${string}`;
+  setStatus: (newStatus: CCTPBridgePhase) => void;
+};
+
+export function useCCTPBridgeStatus({
+  initiateTxHash,
+  bridgeDirection,
+}: UseCCTPBridgeStatusProps): CCTPBridgeStatus {
+  const [status, setStatus] = useState<CCTPBridgePhase>('INITIATE_CCTP_BRIDGE_PENDING');
+  const [message, setMessage] = useState<`0x${string}` | undefined>(undefined);
+  const [messageHash, setMessageHash] = useState<`0x${string}` | undefined>(undefined);
+  const publicClient = usePublicClient({
+    chainId: bridgeDirection === 'deposit' ? l2ChainID : l1ChainID,
```

Comment:
Consider setting `isDeposit = bridgeDirection === 'deposit'` and `chainId = isDeposit ? l2ChainID : l1ChainID` since they seem to come up a few times

Reply:
updated